frontend/src/GameReportSettings.tsx

```diff
@@ -0,0 +1,114 @@
+    settlementLayout: "Standard",
+    areSettlementsAbbreviated: false,
+};
+
+export interface Settings {
+    settlementLayout: SettlementLayout;
+    areSettlementsAbbreviated: boolean;
+}
+
+interface Props {
+    settings: Settings;
+    setSettings: React.Dispatch<React.SetStateAction<Settings>>;
+}
+
+export default function GameReportSettings({ settings, setSettings }: Props) {
+    useEffect(function applyCachedSettings() {
+        setSettings(
+            parseCachedSettings(localStorage.getItem(CACHED_SETTINGS_KEY))
+        );
+    }, []);
+
+    useEffect(
+        function cacheSettings() {
+            localStorage.setItem(CACHED_SETTINGS_KEY, JSON.stringify(settings));
+        },
+        [settings]
+    );
+
+    return (
+        <Box p="10px">
+            <Typography level="h3" mb="10px">
+                Settings
+            </Typography>
+
+            <Box border="1px solid #ccc" borderRadius={"5px"} p="10px">
+                <Typography level="h4" mb="10px">
+                    SP Settlement Captures
+                </Typography>
+
+                <Typography level="body-lg" mb="10px">
+                    Layout
+                </Typography>
+
+                <SingleOptionInput
+                    values={settlementLayouts.slice()}
+                    orientation="vertical"
+                    current={settings.settlementLayout}
+                    validate={() => {}}
+                    onChange={(v) =>
+                        setSettings((prev) => ({
+                            ...prev,
+                            settlementLayout: v,
+                        }))
+                    }
+                />
+
+                <Typography level="body-lg" my="10px">
+                    Names
+                </Typography>
+
+                <SingleOptionInput
+                    values={[false, true]}
+                    getLabel={(v) => (v ? "Abbreviated" : "Full")}
+                    orientation="vertical"
+                    current={settings.areSettlementsAbbreviated}
+                    validate={() => {}}
+                    onChange={(v) =>
+                        setSettings((prev) => ({
+                            ...prev,
+                            areSettlementsAbbreviated: v,
+                        }))
+                    }
+                />
+            </Box>
+        </Box>
+    );
+}
+
+function parseCachedSettings(cachedSettings: string | null): Settings {
+    try {
+        if (cachedSettings) {
+            const parsedSettings: Partial<Settings> | null =
+                JSON.parse(cachedSettings);
+
+            if (isDefined(parsedSettings)) {
+                return {
+                    settlementLayout:
+                        settlementLayouts.find(
+                            (s) => s === parsedSettings.settlementLayout
+                        ) ?? defaultSettings.settlementLayout,
+                    areSettlementsAbbreviated:
```

Comment:
I'm not super psyched about how this function validates schema by naming each property explicitly, but I couldn't get the types to be happy when I tried to make it more generic. It's not a big deal with two properties, but I want to revisit it as the settings get bigger